src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -488,14 +496,86 @@ public FormValidation doCheckTargetCredentialsApiKey(
         }
         if (CredentialsProvider.listCredentials(StringCredentials.class,
                         item,
-                        ACL.SYSTEM, 
+                        ACL.SYSTEM,
                         Collections.emptyList(),
                         CredentialsMatchers.withId(targetCredentialsApiKey)).isEmpty()) {
             return FormValidation.error("Cannot find currently selected credentials");
         }
         return FormValidation.ok();
     }
 
+    /**
+     * Populates the targetCredentialsApplicationKey field from the configuration screen with all of the valid credentials
+     *
+     * @param item - The context within which to list available credentials
+     * @param targetCredentialsApplicationKey - A String containing the application key as a credential
+     * @return a ListBoxModel object used to display all of the available credentials.
+     */
+    public ListBoxModel doFillTargetCredentialsApplicationKeyItems(
+            @AncestorInPath Item item,
+            @QueryParameter("targetCredentialsApplicationKey") String targetCredentialsApplicationKey
+    ) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        // If the users does not have permissions to list credentials, only list the current value
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(targetCredentialsApplicationKey);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                    && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return result.includeCurrentValue(targetCredentialsApplicationKey);
+            }
+        }
```

Comment:
Are there unit tests for the permission logic here?  I might have missed them in the test changes.  Just a recommendation.

My main concern here around tests, outside of good practice for authorization code, is that the logic here is approximately "fail open" (your default state is the sensitive operation) - I only say approximate, because in case of an exception, as far as I know Java will stop execution.

However, this means the logic required some extra thought to figure out exactly the scenarios allowed for access and I worry a little bit about maintainability.

For example, say that Jenkins introduces a new permission type called "SUPERUSER" and we want to allow that too.  Throwing this in requires the DeMorgan-ing our logic to `!Jenkins.get().hasPermission(Jenkins.ADMINISTER) && !Jenkins.get().hasPermission(Jenkins.SUPERUSER)`, but I feel it's easy to make the mistake `!Jenkins.get().hasPermission(Jenkins.ADMINISTER) || !Jenkins.get().hasPermission(Jenkins.SUPERUSER)`, which would be an access control error and grant unrestricted access.

Might get caught in code review, might not - but explicit tests would go a long way here.

Reply:
Agree, that covering this with tests is a good idea, however I would argue that it should be done in a separate PR: in here I am simply introducing an additional field which is handled in a way identical to the others that existed before, and it would be best to change/verify the way permissions are checked in a centralised manner (that is, do it for all config properties).